desktop-src/winmsg/wm-inputlangchange.md

```diff
@@ -48,20 +50,37 @@ An application should return nonzero if it processes this message.
 
 ## Remarks
 
-You can retrieve keyboard [locale name](../Intl/locale-names.md) via [LCIDToLocaleName](/windows/win32/api/winnls/nf-winnls-lcidtolocalename) function. With locale name you can use [modern locale functions](/windows/win32/intl/calling-the--locale-name--functions):
+You can retrieve the [BCP 47](https://www.rfc-editor.org/info/bcp47) [locale name](../Intl/locale-names.md) from the language identifier by calling the [LCIDToLocaleName](/windows/win32/api/winnls/nf-winnls-lcidtolocalename) function. Once you have the locale name, you can then use [modern locale functions](/windows/win32/intl/calling-the--locale-name--functions) to extract additional locale information.
 
 ```cpp
 case WM_INPUTLANGCHANGE:
 {
     HKL hkl = (HKL)lParam;
+    LANGID langId = LOWORD(hkl);
+
     WCHAR localeName[LOCALE_NAME_MAX_LENGTH];
-    LCIDToLocaleName(MAKELCID(LOWORD(hkl), SORT_DEFAULT), localeName, LOCALE_NAME_MAX_LENGTH, 0);
+    LCIDToLocaleName(MAKELCID(langId, SORT_DEFAULT), localeName, LOCALE_NAME_MAX_LENGTH, 0);
 
+    // Get the ISO abbreviated language name (for example, "eng").
     WCHAR lang[9];
     GetLocaleInfoEx(localeName, LOCALE_SISO639LANGNAME2, lang, 9);
 }
 ```
 
+To get the the name of the currently active input locale identifier, call the [GetKeyboardLayoutName](/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew). For more information, see [Languages, Locales, and Keyboard Layouts](/windows/win32/inputdev/about-keyboard-input#languages-locales-and-keyboard-layouts).
+
+For a list of the input layouts that are supplied with Windows, see [Keyboard Identifiers and Input Method Editors for Windows](/windows-hardware/manufacture/desktop/windows-language-pack-default-values).
+
+[Input Method Editor (IME)](/windows/apps/design/input/input-method-editors) profile changes may not be notified with *WM_INPUTLANGCHANGE*. You can use [ITfActiveLanguageProfileNotifySink](/windows/win32/api/msctf/nn-msctf-itfactivelanguageprofilenotifysink) from the [Text Services Framework](/windows/win32/tsf/text-services-framework) to handle active language or text service changes.
+
+**Starting with Windows 8**
+
+Some input layouts may not have assigned language identifiers and could be reported as transient values like `LOCALE_TRANSIENT_KEYBOARD1` (`0x2000`) or `LOCALE_TRANSIENT_KEYBOARD2` (`0x2400`) in the low word of *LPARAM*. 
+
+These values can be re-assigned by the system at any time (such as when the user changes their Language Profile), and because it can mean a different locale for a different user and/or on a different system, it is transient and cannot be used as a durable identifier. See [The deprecation of LCIDs](/globalization/locale/locale-names#the-deprecation-of-lcids) for more info.
```

Comment:
What does "it" refer to in this paragraph?

Reply:
...it should refer to 'transient LANGIDs values' not  just 'These values'.